lib/event-subscribers-builder.js

```diff
@@ -1,10 +1,183 @@
 'use strict';
 
+const path = require('path');
+const YAML = require('yaml');
+
+const EndpointResolver = require('@janiscommerce/endpoint-resolver');
+
+const YmlBuilder = require('@janiscommerce/yml-builder');
+
+const fs = require('./utils/promisified-fs');
+const log = require('./utils/lllog-wrapper');
+
 const EventSubscribersBuilderError = require('./event-subscribers-builder-error');
 
+/**
+ * @class EventSubscribersBuilder
+ * @classdesc Builds event ymls from src then resolve the endpoints
+ */
 class EventSubscribersBuilder {
 
-	// package code...
+	constructor(environment, input, output, schemas) {
+		this.environment = environment;
+		this.input = input;
+		this.output = output;
+		this.schemas = schemas;
+	}
+
+	get inputPath() {
+		return path.join(process.cwd(), this.input);
+	}
+
+	get outputPath() {
+		return path.join(process.cwd(), this.output);
+	}
+
+	get schemasPath() {
+		return path.join(process.cwd(), this.schemas);
+	}
+
+	async _validateBuiltSchemas(schemasPath) {
+
+		try {
+
+			await fs.stat(schemasPath);
+
+		} catch(err) {
+			if(err.code === 'ENOENT') {
+				throw new EventSubscribersBuilderError('Built api schemas not found. Please build before run this utility.',
+					EventSubscribersBuilderError.codes.API_SCHEMAS_NOT_FOUND);
+			}
+			throw new EventSubscribersBuilderError(`Couldn't get the built schemas file: ${err.message}`,
+				EventSubscribersBuilderError.codes.FS_READ_WRITE_ERROR);
+		}
+
+	}
+
+	async _getSourceYml(filepath) {
+
+		try {
+			const sourceFile = await fs.readFile(filepath, 'utf8');
+			return YAML.parse(sourceFile);
+		} catch(err) {
+			throw new EventSubscribersBuilderError(`Unable to get or parse source file data: ${err.message}`,
+				EventSubscribersBuilderError.codes.YML_PARSE_ERROR);
+		}
+	}
+
+	async _resolveEndpoints(environment, schemas, source) {
+
+		if(!Array.isArray(source.events))
+			throw new EventSubscribersBuilderError('Invalid events yml: Should have the events array.', EventSubscribersBuilderError.codes.INVALID_YML);
+
+		const endpointResolver = new EndpointResolver(schemas, environment);
+
+		const resolved = { events: [] };
+
+		for(const event of source.events) {
+
+			if(!Array.isArray(event.listeners))
+				throw new EventSubscribersBuilderError('Invalid event listeners: Should be an array.', EventSubscribersBuilderError.codes.INVALID_YML_LISTENERS);
+
+			const resolvedEvent = { ...event, listeners: [] };
+
+			for(const listener of event.listeners) {
+
+				if(typeof listener.namespace !== 'string' && typeof listener.method !== 'string') {
+					throw new EventSubscribersBuilderError('Invalid event listener: Should be an object with namespace and method properties.',
+						EventSubscribersBuilderError.codes.INVALID_YML_LISTENER_FORMAT);
+				}
+
+				try {
+					const { httpMethod, url } = await endpointResolver.resolve(listener.namespace, listener.method);
+
+					resolvedEvent.listeners.push({
+						namespace: url,
+						method: httpMethod
+					});
```

Comment:
Esto no cumple con lo solicitado en el ticket, se deben pasar unicamente los endpoints porque siempre el método será **POST**.
```suggestion
resolvedEvent.listeners = [...esolvedEvent.listeners, url];
```